casanovo/denovo/model.py

```diff
@@ -444,6 +461,37 @@ def _terminate_finished_beams(
                             abs(d) < self.precursor_mass_tol
                             for d in delta_mass_ppm
                         )
+
+                        # Don't terminate if m/z difference is by NH3 loss
+                        if (
+                            exceeds_precursor_mz_tol
+                            and not is_within_precursor_mz_tol
```

Comment:
**nitpick:** This is always the case if `exceeds_precursor_mz_tol` is True, hence not really needed.

Reply:
They can both be True at the same time (`exceeds_precursor_mz_tol` is True for any mass delta higher than lower bound - so it includes the tolerance range), in which case we wouldn't want to go check for NH3 loss.